appliances/Harbor/appliance.sh

```diff
@@ -0,0 +1,325 @@
+    if [ $? -eq 0 ]; then
+        msg info "Certificate generated successfully"
+    else
+        msg info "Error generating SSL certificate. Resuming..."
+    fi
+}
+
+
+download_unpack_harbor() {
+    msg info "Downloading Harbor..."
+    # Obtain download URL
+    url=$(curl -f -s https://api.github.com/repos/goharbor/harbor/releases/latest \
+    | grep browser_download_url \
+    | cut -d '"' -f 4 \
+    | grep '\.tgz$' \
+    | grep online)
+
+    # Check all pipe exit codes
+    for status in "${PIPESTATUS[@]}"; do
+        if [ "$status" -ne 0 ]; then
+            msg error "Unable to obtain Harbor download URL, aborting..."
+            exit 1
+        fi
+    done
+
+    # Download Harbor
+    wget -P /root/ $url
+    if [ $? -ne 0 ]; then
+	    msg info "Harbor download failed. Download URL: $url"
+        exit 1
+    else
+        msg info "Harbor downloaded successfully"
+    fi
+    tar -xvzf /root/harbor-online-installer-v*.tgz
+    if [ $? -ne 0 ]; then
+	    msg info "Harbor unpacking failed. Aborting..."
+        exit 1
+    else
+        msg info "Harbor unpacked successfully"
+    fi
+    cp /root/harbor/harbor.yml.tmpl /root/harbor/harbor.yml
+}
+
+
+mount_persistent() {
+    # Check if /data directory exists, if not create it
+    if [ ! -d "/data" ]; then
+        mkdir -p /data
+    fi
+
+    # Check if the user provided a persistent device
+    if [ "$HARBOR_PERSISTENT_DEV" = "None" ]; then
+        msg warning "No persistent device specified. Persistency can not be ensured. OS disk will be used to save Harbor data."
+        return 0
+    fi
+
+    # Check if the provided device exists
+    if ! [ -e "/dev/$HARBOR_PERSISTENT_DEV" ]; then
+        msg error "PERSISTENT DEVICE /dev/$HARBOR_PERSISTENT_DEV NOT FOUND. Aborting..."
+    fi
+    msg info "Persistent disk (/dev/$HARBOR_PERSISTENT_DEV) detected"
+
+    # Check if already mounted
+    if $(cat /proc/mounts | grep -qs /dev/$HARBOR_PERSISTENT_DEV); then
+        msg info "/dev/$HARBOR_PERSISTENT_DEV is already mounted."
+        return 0
+    fi
+    msg info "/dev/$HARBOR_PERSISTENT_DEV not yet mounted"
+
+    # Checking if the device contains a filesystem
+
+    if ! blkid -s TYPE -o value /dev/$HARBOR_PERSISTENT_DEV; then
+        msg error "Unable to obtain persistend disk FS type. Please, make sure that the persistent device has a filesystem configured"
+    fi
+
+    # Saving FS type
+    FS_TYPE=$(blkid /dev/$HARBOR_PERSISTENT_DEV --match-tag TYPE -o value)
+    msg info "Detected filesystem type $FS_TYPE for persistent device /dev/$HARBOR_PERSISTENT_DEV"
+
+    # Add entry to /etc/fstab for persistence
+    echo "/dev/$HARBOR_PERSISTENT_DEV   /data   $FS_TYPE   defaults   0   2" | sudo tee -a /etc/fstab > /dev/null
+    msg info "Added entry to /etc/fstab for persistency. Mounting..."
+    # Mount the drive
+    mount -a
+    if [ $? -ne 0 ]; then
+	    msg error "Unable to mount persistent device, aborting..."
```

Comment:
```suggestion
        msg error "Unable to mount persistent device, aborting..."
```

Reply:
Fixed in https://github.com/OpenNebula/one-apps/pull/87/commits/7745f4ab91ecbd2a9bc3632d4ec3bf18ea8c4003